rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java

```diff
@@ -625,6 +625,76 @@ public void logReleaseBtcRequested() {
         Assert.assertArrayEquals(event.encodeEventData(amount.getValue()), result.getData());
     }
 
+    @Test
+    public void logRejectedPegin() {
+        // Setup event logger
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        List<LogInfo> eventLogs = new LinkedList<>();
+
+        BridgeEventLogger eventLogger = new BridgeEventLoggerImpl(null, activations, eventLogs);
+
+        BtcTransaction btcTx = new BtcTransaction(BridgeRegTestConstants.getInstance().getBtcParams());
+
+        eventLogger.logRejectedPegin(btcTx, RejectedPeginReason.PEGIN_CAP_SURPASSED);
+
+        Assert.assertEquals(1, eventLogs.size());
+        LogInfo entry = eventLogs.get(0);
+
+        Assert.assertEquals(PrecompiledContracts.BRIDGE_ADDR, new RskAddress(entry.getAddress()));
+
+        // Assert address that made the log
+        LogInfo result = eventLogs.get(0);
+        Assert.assertArrayEquals(PrecompiledContracts.BRIDGE_ADDR.getBytes(), result.getAddress());
+
+        // Assert log topics
+        Assert.assertEquals(2, result.getTopics().size());
+        CallTransaction.Function event = BridgeEvents.REJECTED_PEGIN.getEvent();
+
+        byte[][] topics = event.encodeEventTopics(btcTx.getHash().getBytes());
+
+        for (int i=0; i<topics.length; i++) {
+            Assert.assertArrayEquals(topics[i], result.getTopics().get(i).getData());
+        }
+
+        // Assert log data
+        Assert.assertArrayEquals(event.encodeEventData(RejectedPeginReason.PEGIN_CAP_SURPASSED.getValue()), result.getData());
```

Comment:
Could we have an explicit test of event having the signature expected by external tools? ie, an assert equals against a calculated hash. So, we could be sure the event is consumible and queryable by external tools, using a code test. My concert is that, if we apply fuzzy testing to the event function code (and then, changing the resulting signature), all these tests could pass in green

Reply:
@ajlopezrsk has a good point here. We are doing this though in the integration tests already, so we can leave this comment as tech debt to be solved by the fed team afterwards